src/thunderbots/software/grsim_communication/grsim_command_primitive_visitor.cpp

```diff
@@ -124,7 +146,34 @@ void GrsimCommandPrimitiveVisitor::visit(
 
 void GrsimCommandPrimitiveVisitor::visit(const KickPrimitive &kick_primitive)
 {
-    // TODO: https://github.com/UBC-Thunderbots/Software/issues/93
+    Point kick_origin    = kick_primitive.getKickOrigin();
+    Angle kick_direction = kick_primitive.getKickDirection();
+
+    double final_speed_at_destination = 0.0;
+    Point perp_line_to_shot           = Point::createFromAngle(kick_direction.quarter());
```

Comment:
I'm assuming you're trying to rotate the angle with `.quarter()`? This won't work because `quarter()` is a `static` function of `Angle` that's used to _create_ angles, not rotate them.

TIL we don't have a `rotate` function for Angles.

To get the Point you're looking for, you can first create a Point from the kick_direction like you're doing (remove the `quarter()`). Then, the `Point` class has a `perp()` function you can use to get the perpendicular Vector/Point. You'll then need to _add_ this Point to the kick_origin. This will give you a Point on the line perpendicular to the direction of the shot (starting from the kick_origin), which matches how you're using this value below.

Make sure you're checking your code compiles. That will help you tell you're on the right path since it will warn you if functions aren't being used as expected.